account_financial_report/wizard/trial_balance_wizard.py

```diff
@@ -183,25 +202,45 @@ def onchange_show_partner_details(self):
         if self.show_partner_details:
             self.receivable_accounts_only = self.payable_accounts_only = True
         else:
-            self.receivable_accounts_only = self.payable_accounts_only = False
+            self.receivable_accounts_only = self.\
+                payable_accounts_only = False
+
+    @api.multi
+    @api.depends('company_id')
+    def _compute_unaffected_earnings_account(self):
+        account_type = self.env.ref('account.data_unaffected_earnings')
+        for record in self:
+            record.unaffected_earnings_account = self.env[
+                'account.account'].search(
+                [
+                    ('user_type_id', '=', account_type.id),
+                    ('company_id', '=', record.company_id.id)
+                ])
+
+    unaffected_earnings_account = fields.Many2one(
+        comodel_name='account.account',
+        compute='_compute_unaffected_earnings_account',
+        store=True
+    )
+
+    @api.multi
+    def _print_report(self, report_type):
+        self.ensure_one()
+        data = self._prepare_report_trial_balance()
+        if report_type == 'xlsx':
+            report_name = 'a_f_r.report_trial_balance_xlsx'
+        else:
+            report_name = 'account_financial_report.trial_balance'
+        return self.env['ir.actions.report'].search(
+            [('report_name', '=', report_name),
+             ('report_type', '=', report_type)], limit=1).report_action(
+            self, data=data)
 
     @api.multi
     def button_export_html(self):
         self.ensure_one()
-        action = self.env.ref(
-            'account_financial_report.action_report_trial_balance')
-        vals = action.read()[0]
-        context1 = vals.get('context', {})
-        if isinstance(context1, pycompat.string_types):
-            context1 = safe_eval(context1)
-        model = self.env['report_trial_balance']
-        report = model.create(self._prepare_report_trial_balance())
-        report.compute_data_for_report()
-
-        context1['active_id'] = report.id
-        context1['active_ids'] = report.ids
-        vals['context'] = context1
-        return vals
+        report_type = 'qweb-html'
+        return self._export(report_type)
```

Comment:
By this change, as is already in the 13.0 version of this module, the report will be shown as HTML in an `iframe` instead of being a direct widget in the screen.

For 13.0, I noticed that HTML does not include the `js` files anymore, so that clicking the values in the report has stopped working. There is an issue about it [here](https://github.com/OCA/account-financial-reporting/issues/690).

I wonder whether including those JS files helps, because they would still be restricted to the iFrame.

Have you thought of a solution for this?